saml/config.go

```diff
@@ -60,55 +158,97 @@ func NewConfig(entityID, acs, issuer, metadata *url.URL) (*Config, error) {
 	return cfg, nil
 }
 
-// GenerateAuthRequestID generates an auth XSD:ID conform ID.
-// A UUID prefixed with an underscore.
-func GenerateAuthRequestID() (string, error) {
-	newID, err := uuid.GenerateUUID()
-	if err != nil {
-		return "", err
-	}
-
-	// Request IDs have to be xsd:ID, which means they need to start with an underscore or letter,
-	// which is not always given for UUIDs.
-	return fmt.Sprintf("_%s", newID), nil
-}
-
 // Validate validates the provided configuration.
 func (c *Config) Validate() error {
 	const op = "saml.Config.Validate"
 
-	if c.AssertionConsumerServiceURL == nil {
-		return fmt.Errorf("%s: ACS URL not set: %w", op, oidc.ErrInvalidParameter)
+	if c.AssertionConsumerServiceURL == "" {
+		return fmt.Errorf("%s: ACS URL not set: %w", op, ErrInvalidParameter)
 	}
-
-	if c.EntityID == nil {
-		return fmt.Errorf("%s: EntityID not set: %w", op, oidc.ErrInvalidParameter)
+	if _, err := url.Parse(c.AssertionConsumerServiceURL); err != nil {
+		return fmt.Errorf("%s: failed to parse ACS URL: %w", op, ErrInvalidParameter)
 	}
 
-	if c.Issuer == nil {
-		return fmt.Errorf("%s: Issuer not set: %w", op, oidc.ErrInvalidParameter)
+	if c.EntityID == "" {
+		return fmt.Errorf("%s: EntityID not set: %w", op, ErrInvalidParameter)
+	}
+	if _, err := url.Parse(c.EntityID); err != nil {
+		return fmt.Errorf("%s: failed to parse EntityID: %w", op, ErrInvalidParameter)
```

Comment:
[non-blocking] I think providing the config params as string makes the UX better. I like that you added parsing the URLs as validation checks here 🚀 
I'm wondering however, if we should change the error message to tell the user that the provided EntityID isn't a valid URL? 🤔

Reply:
Ah yes, that would be helpful 👍 Changing this now.

Reply:
Done in https://github.com/hashicorp/cap/pull/84/commits/3d14e7adb1b29d6b44f764295d4b33d3a6864d31.